core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -2373,6 +2373,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
           s"${KafkaConfig.ControllerListenerNamesProp} must not be empty when running in ZooKeeper migration mode: ${controllerListenerNames.asJava}")
         require(interBrokerProtocolVersion.isMigrationSupported, s"Cannot enable ZooKeeper migration without setting " +
           s"'${KafkaConfig.InterBrokerProtocolVersionProp}' to 3.4 or higher")
+        require(logDirs.size == 1, "Cannot enable ZooKeeper migration when JBOD is in use.")
```

Comment:
nit: I've always disliked the term "JBOD" when in fact it really is multiple log directories.  I know people know that JBOD refers to that, but I would personally prefer if it said "Cannot enable ZooKeeper migration when multiple log directories (aka JBOD) is in use".